File: Source/root/src/main/java/root/model/component/Field.java
package root.model.component;

import java.util.List;

import root.Point2D;
import root.model.PlayerData;
import root.model.component.option.FieldOption;
import root.model.event.ModelEventProducer;

public interface Field extends ModelEventProducer {

	boolean isVisible();

	Unit getUnit();

	void setUnit(Unit newUnit);

	Terrain getTerrain();

	PlayerData getPlayer();

	Point2D getStoragePosition();

	void addFieldOptions(List<FieldOption> options);

	void addFieldOption(FieldOption option);

	List<FieldOption> getEnabledOptions();

	List<FieldOption> adjustOptionsFor(Field secondField);

	List<Point2D> getNeighbours(int r);
}

File: Source/server/rabbit-game/src/Database/Field.cs

namespace RabbitGameServer.Database
{
	public class Field
	{
		public Point2D position;

		public bool isVisible;

		public string unitName;

		public Terrain terrain;

		public bool inBattle;

		public string owner;

		public Field(Point2D position, bool isVisible, string unitName, Terrain terrain, bool inBattle, string owner)
		{
			this.position = position;
			this.isVisible = isVisible;
			this.unitName = unitName;
			this.terrain = terrain;
			this.inBattle = inBattle;
			this.owner = owner;
		}
	}
}